src/main/java/userReg/UserRegUI.java

```diff
@@ -0,0 +1,237 @@
+        frame.add(programOfStudy);
+
+        //Add hobby in frame
+        JComboBox hobby=new JComboBox();
+        //Create sub-item.
+        hobby.addItem("-----SELECT-----");
+        for (Hobbies hb : Hobbies.values()){
+            hobby.addItem(hb);
+
+        }
+        hobby.setBounds(140, 205, 180, 25);
+        frame.add(hobby);
+
+        // Add interests in frame
+        JTextField interestedIn = new JTextField();
+        interestedIn.setText(Gender.MALE.toString());
+        JRadioButton mrb1=new JRadioButton(Gender.MALE.toString(),true);
+        mrb1.setBounds(140,230,90,30);
+        frame.add(mrb1);
+        mrb1.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                interestedIn.setText(mrb1.getText().toString());
+            }
+        });
+        JRadioButton frb1=new JRadioButton(Gender.FEMALE.toString(),false);
+        frb1.setBounds(210,230,90,30);
+        frame.add(frb1);
+        frb.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                interestedIn.setText(frb1.getText().toString());
+            }
+        });
+        ButtonGroup genderBg1=new ButtonGroup();
+        genderBg1.add(mrb1);
+        genderBg1.add(frb1);
+
+        JTextArea selfIntro = new JTextArea(12,34);
+        selfIntro.setBounds(140, 260, 180, 125);
+        frame.add(selfIntro);
+
+        //Set submit button
+        JButton buttonregister = new JButton("Submit");
+        buttonregister.setBounds(160, 450, 100, 25);
+        frame.add(buttonregister);
+
+        // Set register button and process further action including passing data to UserRegUIController.
+        buttonregister.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+
+                String name_v=name.getText();
+                String eMail_v = new String(eMail.getText());
+                String password_v = new String(password.getText());
+                System.out.println(genderText.getText());
+                Gender gender= Gender.valueOf(genderText.getText());
+                int age_v=Integer.valueOf(("".equals(age.getText())?"0":age.getText()));
+                float height_v=Float.valueOf("".equals(height.getText())?"0":height.getText());
+                String programOfStudy_v=programOfStudy.getText();
+                Hobbies hobby_v=Hobbies.valueOf(hobby.getSelectedItem().toString());
+                String selfIntro_v=selfIntro.getText();
+                Gender interestedIn_v=Gender.valueOf(interestedIn.getText());
+                // Initialize
+                User user=new User(name_v, gender, age_v, height_v, programOfStudy_v,
+                        hobby_v, selfIntro_v, interestedIn_v);
+                UserAccount ua= new UserAccount(eMail_v,password_v,user);
```

Comment:
Using User and UserAccount violates clean architecture.  UI should not use classes from entities layer. To fix it: 
Option 1: pass the raw data to controller directly, and let controller create request model for interactor. 
Option 2: if the raw data has too many inputs and thus too many parameters for controller to take,  to avoid smelly code,  create a data structure in the layer of interface adapters, and pass that data structure to controller.

Reply:
Please take a look at the latest version. Thanks ^^.

Reply:
I saw the new version still using the following code which violates the clean architecture. 
 User user=new User(name_v, gender, age_v, height_v, programOfStudy_v,
                        hobby_v, selfIntro_v, interestedIn_v);
                UserAccount ua= new UserAccount(eMail_v,password_v,user);